### ISSUE TITLE
AdSchedule Seems Not Work Properly On Android

### ISSUE THREAD

**Issue by octsiri** [chaimPaneth/react-native-jw-media-player#69]
Hi @chaimPaneth, I am trying to implement VMAP ads on JWPlayer by adding adSchedule on `playlistItem`. But I always get the result that the ad's video does not play and also there is always a loading indicator run forever while the original video won't play. I need your help. Thank you.

This is an example of my playlist item
```
const playlistItem = {
    mediaID: "ykJZQsJn",
    time: 0,
    file: "https://cdn.jwplayer.com/manifests/ykJZQsJn.m3u8",
    controls: true,
    repeat: false,
    adSchedule: [
      {
        tag:
          'https://pubads.g.doubleclick.net/gampad/ads?slotname=/124319096/external/ad_rule_samples&sz=640x480&ciu_szs=300x250&cust_params=deployment%3Ddevsite%26sample_ar%3Dpremidpost&url=https://developers.google.com/interactive-media-ads/docs/sdks/html5/tags&unviewed_position_start=1&output=xml_vast3&impl=s&env=vp&gdfp_req=1&ad_rule=0&vad_type=linear&vpos=preroll&pod=1&ppos=1&lip=true&min_ad_duration=0&max_ad_duration=30000&vrid=6256&video_doc_id=short_onecue&cmsid=496&kfa=0&tfcd=0',
        offset: 'pre',
      },
    ],
```

That case only happens on Android but fine on iOS. 🙏

**Comment by hapo-vupq** [comment]
Hi @octsiri , that is so funny, i run your example on iOS(simulator), Android(real device) (add autostart={true}) and i can see:
- On iOS: not working with mediaID and offset: 'pre' -> change to mediaId and you can see offset setting in https://developer.jwplayer.com/jwplayer/docs/android-schedule-vast-ads ,
scroll to Add multiple ad breaks to a player.
But ADS not working even though there's a dot on the seek bar(offset time).
- On Android: it working, but ADS always runs when the video is ready to run.(not true with offset)
It is my result, and can you tell me why the ADS is running on your iOS device?
Thank you!

**Comment by hapo-vupq** [comment]
Update:
I changed all 'JWAdClientGoogima' to 'JWAdClientVast' in RNJWPlayerNativeView.m and my ADS run on iOS. 😅

**Comment by peter7z** [comment]
@octsiri I can not get the ads to work on iOS. Tried with lots of different tags and offset values, any ideas?
Nothing shows, the video just pays as if it had no ads.

**Comment by hapo-vupq** [comment]
@peter7z @octsiri Have you tried change all 'JWAdClientGoogima' to 'JWAdClientVast' in node_modules/react-native-jw-media-player/ios/RNJWPlayerNativeView.m ?

**Comment by chaimPaneth** [comment]
I've added in the latest version 0.1.48 an adClient prop to be able to determine the ad client it should solve this issue.

**Comment by hapo-vupq** [comment]
@chaimPaneth This problem has been fixed at version 0.1.48.
Thank you.

**Comment by chaimPaneth** [comment]
You welcome! Closing this issue.
